lib/kafka/broker_pool.rb

```diff
@@ -9,7 +9,13 @@ def initialize(connection_builder:, logger:)
     end
 
     def connect(host, port, node_id: nil)
-      return @brokers.fetch(node_id) if @brokers.key?(node_id)
+
+      if @brokers.key?(node_id)
+        broker = @brokers.fetch(node_id)
```

Comment:
I could answer this myself by reading more of the source, but, lazy...does this operation to fetch brokers come back pretty quickly? Do we end up doing it constantly, or does this happen only periodically? Just wondering if this might slow things down or add load on the brokers.

Reply:
1) I am probably missing something, but, isn't the number of times `fetch` is executed unchanged? `return @brokers.fetch(node_id) if @brokers.key?(node_id)` should be equivalent to `if @brokers.key?(node_id); broker = @brokers.fetch(node_id)` in terms of the number of times `fetch` is called?

2) @brokers is a hash, so, `fetch` is pretty fast.

Reply:
Hahah yes, you are right!